lib/Stan_math_code_gen.ml

```diff
@@ -214,20 +285,20 @@ let rec pp_statement ppf {stmt; sloc} =
                   "local_scalar_t__ \
                    DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());" ;
                 text "(void) DUMMY_VAR__;  // suppress unused var warning" ;
-                text "int current_statement_begin__ = -1;" ;
                 pp_located_error ppf (pp_statement, fdbody, None) )
-            , fdbody ) )
+            , fdbody ) ;
+          pf ppf "@ " )
 
 let%expect_test "location propagates" =
-  {sloc= "hi"; stmt= Block [{stmt= Break; sloc= "lo"}]}
+  {sloc= "hi"; stmt= Block [{stmt= NRFunApp ("print", []); sloc= "lo"}]}
   |> strf "@[<v>%a@]" pp_statement
   |> print_endline ;
   [%expect
     {|
-      {
-        current_statement_loc__ = "lo";
-        break;
-      } |}]
+    {
+      current_statement__ = "lo";
+      stan_print(pstream__);
+    } |}]
```

Comment:
Why should the outer location not be printed? Is it set up such that statements always need to be wrapped in blocks for the location to be printed? Note that top level statements are not wrapped in a block. How would we get error locations for those?

Reply:
Locations for Blocks are not printed as all of their member statements have individual locations.